BepInEx.Core/Contract/IPluginLoadContext.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+
+namespace BepInEx;
+
+/// <summary>
+///     An interface that allows to dynamically load and track an assembly
+///     which can be obtained from a provider
+/// </summary>
+public interface IPluginLoadContext : IDisposable
+{
+    /// <summary>
+    ///     An identifier that uniquely identifies an assembly from a provider no matter its revision
+    /// </summary>
+    public string AssemblyIdentifier { get; }
+    
+    /// <summary>
+    ///     An optional hash that changes each time the assembly changes which can be tracked for cache
+    ///     invalidation purposes. If this is null, no caching occurs for this assembly load
+    /// </summary>
+    public string AssemblyHash { get; }
```

Comment:
Is AssemblyHash a proper name if it doesn't have to be a hash?

Reply:
After discussion, it was determined it's fine for it to stay as is.